executor/aggfuncs/func_avg.go

```diff
@@ -181,18 +182,21 @@ func (e *avgOriginal4DistinctDecimal) UpdatePartialResult(sctx sessionctx.Contex
 		if err != nil {
 			return errors.Trace(err)
 		}
-		if isNull || p.valSet.Exist(input) {
+		if isNull {
 			continue
 		}
-
+		decStr := string(hack.String(input.ToString()))
```

Comment:
Why don't use `ToHashKey` here?

Reply:
How about use `ToHashKey` to make `DecimalSet.Exist` correct, otherwise `DecimalSet` is useless...

Reply:
I'll change it to ToHashKey.